### ISSUE TITLE
Usage Question: Customized Data field based on columns data in row.

### ISSUE THREAD

**Issue by stixoffire** [volkanceylan/Serenity#1534]
I have a Column that is made up of the selections from multiple columns and a sequence number in a separate table. Upon use of the sequence number I also need to increment it.
Basically I have orderNumber column = SalesPersonCode + 2 Digit Year + [Dash] + SequenceNumber
There are 3 tables involved and a brief structure is as follows:

**tblOrders:**
orderId
orderNumber 

> [Business rule requires this and it is this field that I need to concatenate with values
> Please note SequenceNumber can not be duplicated. So 2 users cannot simply retrieve it and concatenate - it has to be done on save and then increment the sequence. The Sequence number will be reset annually .]

_SalesPersonID_  [Foreign Key - to SalesPersonId in tblSalesPerson]

**tblSalesPerson:**
SalesPersonID
SalesPersonName
SalesPersonCode

**tblApplicationData:**
SequenceNumber

When I am saving the order I want to get the SequenceNumber - and update the orderNumber  based on the columns and its value, and after the save executes the tblApplicationData.SequenceNumber must be incremented.

> What is the best approach to accomplish this in Serenity ?

**Comment by edson** [comment]
Hi, @stixoffire . I have done this in repository, in MySaveHandler, overriding 

` protected override void BeforeSave()`

I just get the sequence number and update it... Doing the mix afterwards in Row.field, etc...
Anyway... Keep in mind that you may have to deal with concurrency...

Best Regards

**Comment by stixoffire** [comment]
@edson Hello and thank you - I am in my repository and inside the MySaveHandler {} I went to override the method - so I have this                    
```
                    string dateYear = DateTime.Now.ToString("yy");
                    string saleID = fld.SalesPersonIdentifier.ToString();
                    MyAppDataRetrieveHandler appData = new MyAppDataRetrieveHandler();
```
I am stuck on the query - I want to get the last sequence (there is only one row ever in this table and only one value LastSequenceNumber 
                  `  appData.Query`

I have since also looked at doing this (but I am kind of stuck as I see no examples of this):
```
                    var query = new SqlQuery();
                    query.Select("LastAssignedQuoteSequence");
                    query.From("ApplicationData");
                    query.OrderBy("Id");

                    query.ExecuteReader(Connection);

                    string quotingNumber = saleID + dateYear + "-" + "";

                    fld.QuoteNumber.DefaultValue = quotingNumber;

                    Request.Entity.QuoteNumber = quotingNumber;
```

**Comment by edson** [comment]
Hi, @stixoffire .

Ok... Let's make things simple... Do you have generated or created a tblApplicationDataRow.cs, endpoints, repository?

If so, you could do something like this in your OrderRepository, MySaveHandler to get the value:
 ```
    if(IsCreate) { /*Only do this when it is a new Order to update orderNumber...*/
                  var aFld = tblApplicationDataRow.Fields;

                   var myApplicationDataRow = Connection.TryFirst<tblApplicationDataRow>(q => q
                            .Select(aFld.SequenceNumber)
                            );

                        if(myApplicationDataRow != null) {
                               int sequenceNumber = myApplicationDataRow.SequenceNumber;
                             myApplicationDataRow.SequenceNumber++;

                             new tblApplicationDataRepository().Update(this.UnitOfWork, new SaveRequest<tblApplicationDataRow>
                            {
                                Entity = myApplicationDataRow
                            });
                           /*or use connection.UpdateById  to update Sequence... */
                               string dateYear = DateTime.Now.ToString("yy");
                               Row.OrderNumber = ....
                        }
....
```

Please, note that I didn't check all syntax and don't know your field types...
There is some code in samples or Northwind for repositories as far as I remember. But I'm in a rush here... Be back later.

Hope it helps...

Best Regards

**Comment by stixoffire** [comment]
@edson All is generated - I will look to the Northwind examples and your code sample - Big THX..
The field is required in my DB (not nullable), and on my Form for validation the user has an input - which I will need to remove validation requirement and somehow prevent the user from entering there - so the save can take place and the value inserted. I will look to the Northwind examples if I can find something like this - I had been searching but maybe I missed something.

**Comment by stixoffire** [comment]
@edson Your suggestions and guidance worked perfectly - I created a separate static class for my Generated OrderNumber and I simply call it. In the BeforeSave, I then update the SequenceNumber in an overidden execute save method - there are not many users <10 and the chance for a concurrency issue of a duplicate is slim. I will think on how better to do this if need be. THX -again - I searched Northwind for the samples but yours seemed much clearer to me,
